include/intersectables/intersectable.h

```diff
@@ -4,8 +4,21 @@
 #include "../hitRecord.h"
 #include "../ray.h"
 
+class BoundingBox;
+
 class Intersectable {
  public:
-  virtual HitRecord* intersect(const Ray&) const = 0;
+  /**
+   * Implement ray-surface intersection in this method. Implementations of this
+   * method need to make and return a {@link HitRecord} correctly, following
+   * the conventions assumed for {@link HitRecord}.
+   *
+   * @param ray the ray used for intersection testing
+   * @return a hit record, should return an invalid hit record if there is no
+   *   intersection
+   */
+  virtual HitRecord* intersect(const Ray& ray) const = 0;
+
+  virtual BoundingBox getBoundingBox() const = 0;
```

Comment:
Ideally, we would return a const reference here. Otherwise, we would always call BoundingBox' copy constructor. 

Unfortunately, this does not work for two reasons: the class `BoundingBox` cannot return **this** this way and for the `Mesh` class we cannot determine the **aabb** in Mesh constuctor's initializer (since the mesh is formed within its constructor's body).

Reply:
You can dereference `this` for returning as const ref, afaik:
```cpp
const BoundingBox& getBB() {
  return *this
}
```